packages/assets/src/loader/Loader.ts

```diff
@@ -206,9 +250,28 @@ export class Loader
         await Promise.all(promises);
     }
 
-    /** All loader parsers registered */
-    public get parsers(): LoaderParser[]
+    /** validates our parsers, right now it only checks for name conflicts but we can add more here as required! */
+    private _validateParsers()
     {
-        return this._parsers;
+        this._parsersValidated = true;
+
+        this._parserHash = {};
+
+        for (let i = 0; i < this._parsers.length; i++)
+        {
+            const parser = this._parsers[i];
+
+            if (parser.name)
+            {
+                if (this._parserHash[parser.name])
+                {
+                    // #if _DEBUG
+                    console.warn(`[Assets] loadParser name conflict "${parser.name}"`);
+                    // #endif
+                }
+
+                this._parserHash[parser.name] = parser;
+            }
+        }
```

Comment:
Could we do something like this instead?

```suggestion
        this._parserHash = this._parsers
            .filter(parser => parser.name)
            .reduce((hash, parser) =>
            {
                if (hash[parser.name])
                {
                    // #if _DEBUG
                    console.warn(`[Assets] loadParser name conflict "${parser.name}"`);
                    // #endif
                }
                
                return { ...hash, [parser.name]: parser };
            }, {});
```